circe/pricing_push/compute.py

```diff
@@ -774,46 +793,71 @@ def process_IN_CLOSE_WRITE(self, event):
+            logging.debug('Sending the output files to the corresponding destinations')
             if flag=='true': 
-                print('send a single output of the task to all its children') 
+                logging.debug('send a single output of the task to all its children') 
+
+                #Unicast
+                # destinations = ["/centralized_scheduler/input/" +x + "/"+home_id+"/"+new_file for x in next_tasks_map[task_name]]
+                # for idx,host in enumerate(next_hosts):
+                #     if self_ip!=combined_ip_map[host]: # different node
+                #         transfer_data(host,username,password,event.pathname, destinations[idx])
+                #     else: # same node
+                #         copyfile(event.pathname, destinations[idx])
+
+                # Using multicast
+                logging.debug('Using multicast instead')
                 destinations = ["/centralized_scheduler/input/" +x + "/"+home_id+"/"+new_file for x in next_tasks_map[task_name]]
-                for idx,host in enumerate(next_hosts):
-                    if self_ip!=combined_ip_map[host]: # different node
-                        transfer_data(host,username,password,event.pathname, destinations[idx])
-                    else: # same node
-                        copyfile(event.pathname, destinations[idx])
+                users = [username]*len(destinations)
+                passwords = [password]*len(destinations)
+                sources = [event.pathname]*len(destinations)
+                transfer_multicast_data(next_hosts,users,passwords,sources, destinations)
+
             else:
                 #it will wait the output files and start putting them into queue, send frst output to first listed child, ....
-                print('wait for all the output files')
+                logging.debug('wait for all the output files')
                 if key not in files_mul:
                     files_mul[key] = [event.pathname]
                 else:
                     files_mul[key] = files_mul[key] + [event.pathname]
                 if len(files_mul[key]) == len(next_hosts):
-                    print('Enough output files to transfer')
-                    for idx,host in enumerate(next_hosts):
-                        current_file = files_mul[key][idx].split('/')[-1]
-                        destinations = "/centralized_scheduler/input/" +next_tasks_map[task_name][idx]+"/"+home_id+"/"+current_file
-                        if self_ip!=combined_ip_map[host]:
-                            print('transfer file to remote node')
-                            transfer_data(host,username,password,files_mul[key][idx], destinations)
-                        else: 
-                            print('copy file')
-                            copyfile(files_mul[key][idx],destinations)
+                    logging.debug('Enough output files to transfer')
+
+                    #Unicast
```

Comment:
Remove commented code.

Reply:
Same reason above for unicast/ multicast.